src/ray/raylet/node_manager.cc

```diff
@@ -2440,6 +2441,30 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &node_stats_
   }
 }
 
+void NodeManager::HandleGetResourceUsageByTask(const rpc::GetResourceUsageByTaskRequest &request,
+                                               rpc::GetResourceUsageByTaskReply *reply,
+                                               rpc::SendReplyCallback send_reply_callback) {
+  // We get the logical resources acquired by each worker that the raylet has leased out
+  RAY_LOG(INFO) << "HandleGetResourceUsageByTask: lease workers: " << leased_workers_.size();
+  for (const auto &worker_it : leased_workers_) {
+    rpc::TaskResourceUsage task_resource_usage;
```

Comment:
Can you add an unit test for this part?

Reply:
We agreed that unittest is not needed for handlers as it is tested in e2e tests.